packages/fxa-settings/src/lib/context/implementations/storage-context.ts

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Context } from "..";
+
+export class StorageContext implements Context {
+
+  private static readonly NAMESPACE = '__fxa_session';
+  private static readonly PERSIST_TO_LOCAL_STORAGE = ['oauth'];
+
+  private state: Record<string, any>;
+
+  constructor(
+    private window: Window
+  ) {
+    this.state = {};
+    this.load();
+  }
+
+  public requiresSync() {
+    return true;
+  }
+
+  public load() {
+    let values = {};
```

Comment:
This was ported from content server. I am not in love with the way this works. I think the JSON serialization adds unnecessary overhead, and we could just use storage like a keystore which would be more performant, more direct, and possibly not require concerns about load and persist. 

At this point, we have to maintain backwards compatibility, but it might be interesting to revisit this.

Reply:
I just looked at the storage.ts class more closely. I like this a lot better and it should conform with Context interface, so I'll try switching to this. Thanks for the call out 👍. Probably no need for a follow up. If the namespace proves to a problem, I am sure it can worked around if necessary.

Reply:
@LZoog I ended up filing a follow up for this. I started to look into it but don't want to miss the train. See FXA-6801.